storedata.js

```diff
@@ -0,0 +1,34 @@
+
+if(!localStorage.getItem('email')) {
+  storeData();
+} else {
+  displayData();
+}
+
+function storeData() {
+  localStorage.setItem('firstname', JSON.stringify(document.getElementById('firstname').value));
+  localStorage.setItem('lastname', JSON.stringify(document.getElementById('lastname').value));
+  localStorage.setItem('fullname', JSON.stringify(document.getElementById('name').value));
+  localStorage.setItem('email', JSON.stringify(document.getElementById('email').value));
+  localStorage.setItem('msg', JSON.stringify(document.getElementById('msg').value));
+  
+  displayData();
+}
+
+function displayData() {
+  document.getElementById('firstname').value = localStorage.getItem('firstname');
+  document.getElementById('lastname').value = localStorage.getItem('lastname');
+  document.getElementById('name').value = localStorage.getItem('fullname');
+  document.getElementById('email').value = localStorage.getItem('email');
+  document.getElementById('msg').value = localStorage.getItem('msg');
+}
+
+document.getElementById('email').addEventListener('change', () => { storeData(); });
+
+document.getElementById('firstname').addEventListener('change', () => { storeData(); });
+
+document.getElementById('lastname').addEventListener('change', () => { storeData(); });
+
+document.getElementById('name').addEventListener('change', () => { storeData(); });
+
+document.getElementById('msg').addEventListener('change', () => { storeData(); });
```

Comment:
- [ ] Great job working on your project. However, when I open the project, I realize that some unwanted characters are being entered even without me inputting any data into the form input fields themselves.

**This is what I am seeing including the localstorage**

![image](https://user-images.githubusercontent.com/86318284/202697864-210adc6a-e777-45a0-8357-4286a034530d.png)

**My view helping you do this neater**

For you to resolve this, you need to have an object which is going to hold your data you could do something like this:

```

let localSt = {
  FullName: '',
  Email: '',
  message: '',
};
```

Then, you now need to have a way to keep track ofwhen the changes are made and also track when the page reloads, the data is being persisted. This you could do it like this:

```
if (localStorage.getItem('localSt') != null) {
  localSt = JSON.parse(localStorage.getItem('localSt'));
  contactForm.FullName.value = localSt.FullName;
  contactForm.Email.value = localSt.Email;
  contactForm.message.value = localSt.message;
}
contactForm.addEventListener('change', () => {
  localSt.FullName = contactForm.FullName.value;
  localSt.Email = contactForm.Email.value;
  localSt.message = contactForm.message.value;
  localStorage.setItem('localSt', JSON.stringify(localSt));
});
```

**Kindly look into this and ensure that the data being stored is clean and void of the un wanted characters that have been added without me adding them.**